emails/views.py

```diff
@@ -270,6 +270,31 @@ def _sns_notification(json_body):
     return _sns_message(message_json)
 
 
+def _get_relay_recipient_from_message_json(message_json):
+    # Go thru all To, Cc, and Bcc fields and
+    # return the one that has a Relay domain
+
+    # First check commmon headers for to or cc match
+    headers_to_check = 'to', 'cc'
+    domains_to_check = get_domains_from_settings().values()
+    common_headers = message_json['mail']['commonHeaders']
+    for header in headers_to_check:
+        if header in common_headers:
+            for recipient in common_headers[header]:
+                for domain in domains_to_check:
+                    if domain in recipient:
+                        return parseaddr(recipient)[1]
+
+    # SES-SNS sends bcc in a different part of the message
+    recipients = message_json['receipt']['recipients']
+    for recipient in recipients:
+        for domain in domains_to_check:
+            if domain in recipient:
+                return recipient
```

Comment:
We should separate this out to its own function since it's being duplicated in two places. A good name could be: `_get_recipient_with_relay_domains`

Reply:
It's not exactly duplicated. 😢 In the case of `commonHeaders` we `return parseaddr(recipient)[1]` but in the case of `recipients` we `return recipient`

Reply:
Oh, I didn't see that 🤦 but that means we can do something like:
```
def _get_recipient_with_relay_domains
    for recipient in recipients:
        for domain in domains_to_check:
            if domain in recipient:
                return recipient
    return None
```
and in `SES-SNS sends bcc in a different part of the message` block `return recipient` and in `commmon headers for to or cc match` we can `return parseaddr(recipient)[1] if recipient else recipient`